xml_analyzer/data_container.py

```diff
@@ -1,38 +1,72 @@
+    autocalc = 'autocalc'
 
 
 class DataContainer:
     def __init__(self):
-        self.mapping_element = collections.namedtuple('property_value_mapping', ['property', 'value'])
-        self.mappings = dict()
-        self.mappings['variables'] = dict()
-        self.result_property = ''
-        self.dependencies = list()
-        self.constraint_specification = ''
+        self._mapping_element = MappingElement
+        self._mappings = dict()
+        self._mappings[DataContainerFields.variables] = dict()
+        self._mappings[DataContainerFields.autocalc] = dict()
+        self._result_property = ''
+        self._dependencies = list()
+        self._constraint_specification = ''
 
     def add_prop_val_mapping(self, prop: str, value: str):
-        map_element = self.mapping_element(prop, value)
-        self.mappings['variables'][prop] = value
+        self._mappings[DataContainerFields.variables][prop] = float(value)
 
-    def add_result_property(self, prop):
-        self.result_property = prop
+    @property
+    def prop_val_mappings(self) -> dict:
+        return self._mappings[DataContainerFields.variables]
 
-    def add_auto_calc_mapping(self, sysml_type_name: str, autocalc_method: str, values: list):
-        self.mappings['autocalc'] = dict()
-        map_element = self.mapping_element(sysml_type_name, values)
-        self.mappings['autocalc'][autocalc_method] = map_element
+    @property
+    def autocalc_mappings(self) -> dict:
+        return self._mappings[DataContainerFields.autocalc]
 
-    def add_dependencies(self, dependencies: list):
-        self.dependencies = dependencies
+    @property
+    def result_property(self) -> str:
+        return self._result_property
 
-    def set_constraint_specification(self, constraint_spec: str):
-        self.constraint_specification = constraint_spec
+    @result_property.setter
+    def result_property(self, prop):
+        self._result_property = prop
 
-    def get_variable_names(self) -> list:
-        return self.mappings['variables'].keys()
+    @property
+    def constraint_specification(self):
+        return self._constraint_specification
+
+    @constraint_specification.setter
+    def constraint_specification(self, constraint_spec: str):
+        self._constraint_specification = constraint_spec
+
+    @property
+    def dependencies(self):
+        return self._dependencies
 
-    def get_dependencies(self):
-        return self.dependencies
+    @dependencies.setter
+    def dependencies(self, dependencies: list):
+        self._dependencies = dependencies
+
+    def add_dependency_mapping(self, prop: str):
+        self._mappings[DataContainerFields.variables][prop] = 'dep'
+
+    def add_auto_calc_mapping(self, sysml_type_name: str, autocalc_method: str, values: list):
+        try:
+            map_elements = self._mappings[DataContainerFields.autocalc][autocalc_method]
+        except KeyError:
+            map_elements = list()
+        finally:
```

Comment:
Everything in the finally block gets executed, even if a exception is raised. Is this intended? If you want to do something if no exception was raised, then use the else block

Reply:
I am using the KeyError exception to trigger the creation of an empty list in case there is not already an existing one for this specific key.... maybe this is a little too brutal?

Reply:
Thats fine, it just looked unintended